connect_ext_ppr/service.py

```diff
@@ -90,3 +97,136 @@ def update_product(data, config, logger):
             product.version = data['version']
             db.add(product)
             db.commit()
+
+
+def get_ppr_new_version(db, deployment):
+    version = (
+        db.query(PPRVersion.version)
+        .filter_by(deployment=deployment.id)
+        .order_by(desc(PPRVersion.version))
+        .with_for_update()
+        .limit(1)
+        .scalar()
+    ) or 0
+    new_version = version + 1
+    return new_version
+
+
+def create_ppr(ppr, context, deployment, db, client, logger):
+    file_data = ppr.file
+    new_version = get_ppr_new_version(db, deployment)
+    config_kwargs = {}
+    status = PPRVersion.STATUS.ready
+    if not file_data:
+        active_configuration = (
+            db.query(Configuration)
+            .filter_by(
+                deployment=deployment.id,
+                state=Configuration.STATE.active,
+            ).first()
```

Comment:
Better to use .one_or_none(), I suppose, to catch situation if there are more then one active configuration.